src/backend/src/api/endpoints/deployment.rs

```diff
@@ -89,16 +102,152 @@ impl DeploymentsEndpoints {
 
     pub async fn create_deployment(
         &mut self,
-        calling_principal: &Principal,
+        calling_principal: Principal,
         sdl: String,
     ) -> Result<DeploymentId, ApiError> {
         self.access_control_service
-            .assert_principal_not_anonymous(calling_principal)?;
+            .assert_principal_not_anonymous(&calling_principal)?;
 
-        let user_id = UserId::new(*calling_principal);
+        let parsed_sdl = SdlV3::try_from_str(&sdl)
+            .map_err(|e| ApiError::invalid_argument(&format!("Invalid SDL: {}", e)))?;
+
+        let user_id = UserId::new(calling_principal);
 
-        let deployment = Deployment::new(sdl, user_id);
+        let deployment_id = self
+            .deployments_service
+            .init_deployment(user_id, sdl)
+            .await?;
 
-        self.deployments_service.create_deployment(deployment).await
+        ic_cdk_timers::set_timer(Duration::from_secs(0), move || {
+            ic_cdk::spawn(async move {
+                if let Err(e) =
+                    handle_deployment(calling_principal, parsed_sdl, deployment_id).await
+                {
+                    print(&format!("Error handling deployment: {:?}", e));
```

Comment:
why aren't we setting the state of the deployment as failed here? Or in any of the handle_* functions. When we deploy on mainnet, `print` statements are useless